zone/mob.cpp

```diff
@@ -1692,151 +1694,77 @@ void Mob::SendIllusionPacket(
+	uint8 new_haircolor;
+	uint8 new_beardcolor;
+	uint8 new_eyecolor1;
+	uint8 new_eyecolor2;
+	uint8 new_hairstyle;
+	uint8 new_luclinface;
+	uint8 new_beard;
+	uint8 new_aa_title;
+	uint32 new_drakkin_heritage;
+	uint32 new_drakkin_tattoo;
+	uint32 new_drakkin_details;
+
+	race = (in_race) ? in_race : GetBaseRace();
+
+	if (in_gender != 0xFF)
+		{
+		gender = in_gender;
 		}
-		else {
-			gender = in_gender;
+	else
+		{
+		gender = (in_race) ? GetDefaultGender(race, gender) : GetBaseGender();
 		}
-	}
 
-	if (in_texture == 0xFF) {
-		if (IsPlayerRace(in_race)) {
-			texture = 0xFF;
-		}
-		else {
-			texture = GetTexture();
+	if (in_texture == 0xFF && !IsPlayerRace(in_race))
+		{
+		new_texture = GetTexture();
 		}
-	}
-	else {
-		texture = in_texture;
-	}
 
-	if (in_helmtexture == 0xFF) {
-		if (IsPlayerRace(in_race)) {
-			helmtexture = 0xFF;
-		}
-		else if (in_texture != 0xFF) {
-			helmtexture = in_texture;
-		}
-		else {
-			helmtexture = GetHelmTexture();
+	if (in_helmtexture == 0xFF && !IsPlayerRace(in_race))
+		{
+		new_helmtexture = GetHelmTexture();
 		}
-	}
-	else {
-		helmtexture = in_helmtexture;
-	}
 
-	if (in_haircolor == 0xFF) {
-		haircolor = GetHairColor();
-	}
-	else {
-		haircolor = in_haircolor;
-	}
-
-	if (in_beardcolor == 0xFF) {
-		beardcolor = GetBeardColor();
-	}
-	else {
-		beardcolor = in_beardcolor;
-	}
-
-	if (in_eyecolor1 == 0xFF) {
-		eyecolor1 = GetEyeColor1();
-	}
-	else {
-		eyecolor1 = in_eyecolor1;
-	}
-
-	if (in_eyecolor2 == 0xFF) {
-		eyecolor2 = GetEyeColor2();
-	}
-	else {
-		eyecolor2 = in_eyecolor2;
-	}
-
-	if (in_hairstyle == 0xFF) {
-		hairstyle = GetHairStyle();
-	}
-	else {
-		hairstyle = in_hairstyle;
-	}
-
-	if (in_luclinface == 0xFF) {
-		luclinface = GetLuclinFace();
-	}
-	else {
-		luclinface = in_luclinface;
-	}
-
-	if (in_beard == 0xFF) {
-		beard = GetBeard();
-	}
-	else {
-		beard = in_beard;
-	}
-
-	aa_title = in_aa_title;
-
-	if (in_drakkin_heritage == 0xFFFFFFFF) {
-		drakkin_heritage = GetDrakkinHeritage();
-	}
-	else {
-		drakkin_heritage = in_drakkin_heritage;
-	}
-
-	if (in_drakkin_tattoo == 0xFFFFFFFF) {
-		drakkin_tattoo = GetDrakkinTattoo();
-	}
-	else {
-		drakkin_tattoo = in_drakkin_tattoo;
-	}
-
-	if (in_drakkin_details == 0xFFFFFFFF) {
-		drakkin_details = GetDrakkinDetails();
-	}
-	else {
-		drakkin_details = in_drakkin_details;
-	}
-
-	if (in_size <= 0.0f) {
-		size = GetSize();
-	}
-	else {
-		size = in_size;
-	}
+	new_haircolor = (in_haircolor == 0xFF) ? GetHairColor() : in_haircolor;
+	new_beardcolor = (in_beardcolor == 0xFF) ? GetBeardColor() : in_beardcolor;
+	new_eyecolor1 = (in_eyecolor1 == 0xFF) ? GetEyeColor1() : in_eyecolor1;
+	new_eyecolor2 = (in_eyecolor2 == 0xFF) ? GetEyeColor2() : in_eyecolor2;
+	new_hairstyle = (in_hairstyle == 0xFF) ? GetHairStyle() : in_hairstyle;
+	new_luclinface = (in_luclinface == 0xFF) ? GetLuclinFace() : in_luclinface;
+	new_beard = (in_beard == 0xFF) ? GetBeard() : in_beard;
+	new_drakkin_heritage = 
```

Comment:
Needs to be 0xFFFFFFFF

Reply:
I think I got them all now.  Just drakkin stuff right?  The others are 0xFF everywhere.